primitives/core/src/offchain/testing.rs

```diff
@@ -156,10 +158,12 @@ impl OffchainState {
 	}
 
 	fn fulfill_expected(&mut self, id: u16) {
-		if let Some(mut req) = self.expected_requests.remove(&RequestId(id)) {
-			let response = req.response.take().expect("Response checked while added.");
+		if let Some(mut req) = self.expected_requests.remove(&RequestId(self.request_ctr)) {
+			let response = req.response.take().expect("Response checked when added.");
 			let headers = std::mem::take(&mut req.response_headers);
 			self.fulfill_pending_request(id, req, response, headers);
+
+			self.request_ctr = self.request_ctr.saturating_add(1);
```

Comment:
```suggestion
			self.request_ctr = self.request_counter.wrapping_add(1);
```
I think wrapping add might make more sense, if the datatype is saturated it is likely it's a long running test, so the initial ones are very likely to not be there anymore. Or just panic.

Reply:
My thinking was this: the max number of requests allowed by the lib is `u16::MAX` so if you get there your test will fail regardless of what is going on here; might as well stop incrementing at that point. 
Now that I think about it I think this should panic.